google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java

```diff
@@ -384,6 +385,34 @@ public void testDeleteRecursive() throws IOException {
     }
   }
 
+  @Test
+  public void testListFilesInRootDirectory() throws IOException {
+    CloudStorageFileSystem fs = CloudStorageFileSystem.forBucket(
+      BUCKET, CloudStorageConfiguration.builder().permitEmptyPathComponents(true)
+      .build());
+    
+    // test absolute path
+    Path rootPath = fs.getPath("");
+    List<String> objectNames = new ArrayList<String>();
+    for (Path path : Files.newDirectoryStream(rootPath)) {
+      objectNames.add(path.toString());
+    }
+    Collections.sort(objectNames);
+    assertThat(objectNames.size()).isEqualTo(2);
+    assertThat(objectNames.get(0)).isEqualTo(BIG_FILE);
+    assertThat(objectNames.get(1)).isEqualTo(SML_FILE);
```

Comment:
You don't need to sort. Try `assertThat(objectNames).containsExactly(BIG_FILE, SML_FILE)`